.github/workflows/bump.yaml

```diff
@@ -0,0 +1,47 @@
+---
+name: Bump
+
+on:
+  push:
+    branches:
+      - '**'
+  workflow_dispatch: # Manual workflow trigger
+
+jobs:
+  bump-so-version:
+    name:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}
+    steps:
+      - name: Setup Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.20.x
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Always checkout main, we don't support bumping for patch releases (for now)
+          ref: main
+          path: ./src/github.com/${{ github.repository }}
+          fetch-depth: 0
+
+      - name: Bump SO
+        working-directory: ./src/github.com/${{ github.repository }}
+        run: go run hack/cmd/bumpso/bumpso.go
+
+      - name: Regenerate all generated files
+        working-directory: ./src/github.com/${{ github.repository }}
+        run: make generated-files
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          path: ./src/github.com/${{ github.repository }}
+          branch: auto/bump-so-version-${{ env.SO_VERSION }}
```

Comment:
Is the `$SO_VERSION` exported somewhere?

Reply:
I removed this usage, I believe there is no easy way to get the branch that triggered the pull request / to which version we're bumping at this stage of the workflow, so we will use the target branch